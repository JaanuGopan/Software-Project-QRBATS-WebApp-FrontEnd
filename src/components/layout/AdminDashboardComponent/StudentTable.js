import React, { useState, useEffect } from "react";
import "../../../pages/AdminDashboard/AdminDashboard.css";
import { FaEdit } from "react-icons/fa";
import GetAllStudentsService from "../../../api/services/GetAllStudentService";

const StudentTable = ({ handleUpdateStudent }) => {
  const [students, setStudents] = useState([]);

  useEffect(() => {
    // Fetch the list of students from the API
    const response = GetAllStudentsService.getAllStudent();
    response.then((res) => {
      setStudents(res);
    });
    console.log(students);
  }, []);

  const deparmentList = ["DEIE", "DCOM", "DMME", "DCEE", "DMENA"];

  return (
    <div className="tableDesign">
      <table className="student-tableArrangement">
        <thead>
          <tr>
            <th>No</th>
            <th className="expand">Name</th>
            <th>IndexNo</th>
            <th>Department</th>
            <th>Username</th>
            <th>Semester</th>
            <th>Edit</th>
          </tr>
        </thead>
        <tbody>
          {students.map((student, index) => (
            <tr key={index}>
              <td>{index + 1}</td>
              <td>{student.studentName}</td>
              <td>{student.indexNumber}</td>
              <td>{deparmentList[student.departmentId - 1]}</td>
              <td>{student.username}</td>
              <td>{student.currentSemester}</td>
              <td>
                <button
                  onClick={() => handleUpdateStudent(student)}
                  className="EditButton"
                >
                  <FaEdit className="EditIcon" />
                </button>
              </td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default StudentTable;
