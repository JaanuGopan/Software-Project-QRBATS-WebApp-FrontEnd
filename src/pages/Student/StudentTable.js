import React, { useState, useEffect } from "react";
import "../AdminDashboard/AdminDashboard.css";
import { FaEdit } from "react-icons/fa";

const StudentTable = ({ handleUpdateStudent, onStudentClick, studentList }) => {
  const [selectedStudent, setSelectedStudent] = useState(null);

  const handleStudentClick = (student) => {
    onStudentClick(student);
    setSelectedStudent(student);
  };

  const departmentList = ["DEIE", "DCOM", "DMME", "DCEE", "DMENA", "DIS"];

  return (
    <div className="tableDesign">
      <table className="student-tableArrangement">
        <thead>
          <tr>
            <th>No</th>
            <th className="expand">Name</th>
            <th>IndexNo</th>
            <th>Department</th>
            <th>Semester</th>
            <th>Edit</th>
          </tr>
        </thead>
        <tbody>
          {studentList.map((student, index) => (
            <tr
              key={index}
              onClick={() => handleStudentClick(student)}
              className={
                selectedStudent === student ? "selected-row" : "event-row"
              }
            >
              <td>{index + 1}</td>
              <td>{student.studentName}</td>
              <td>{student.indexNumber}</td>
              <td>{departmentList[student.departmentId - 1]}</td>
              <td>{student.currentSemester}</td>
              <td>
                <button
                  onClick={() => handleUpdateStudent(student)}
                  className="EditButton"
                >
                  <FaEdit className="EditIcon" />
                </button>
              </td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default StudentTable;
