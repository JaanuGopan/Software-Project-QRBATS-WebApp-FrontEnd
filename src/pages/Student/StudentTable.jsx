import React, { useState, useEffect } from 'react';
import '../AdminDashboard/AdminDashboard.css';
import { FaEdit } from 'react-icons/fa';

const StudentTable = ({ handleUpdateStudent, onStudentClick, studentList, search }) => {
  const [selectedStudent, setSelectedStudent] = useState(null);
  
  const handleStudentClick = (student) => {
    onStudentClick(student);
    setSelectedStudent(student);
  };

  const departmentList = ['DEIE', 'DCOM', 'DMME', 'DCEE', 'DMENA', 'DIS'];

  return (
    <div className="tableDesign">
      <table className="student-tableArrangement">
        <thead>
          <tr>
            <th>No</th>
            <th className="expand">Name</th>
            <th>IndexNo</th>
            <th>Department</th>
            <th>Semester</th>
            <th>Edit</th>
          </tr>
        </thead>
        {studentList.filter(
              (student) => student.firstName.toLowerCase().includes(search.toLowerCase()) ||
              student.indexNumber.toLowerCase().includes(search.toLowerCase())
            ).length > 0 ? <tbody>
          {studentList.filter(
              (student) => student.firstName.toLowerCase().includes(search.toLowerCase()) ||
              student.indexNumber.toLowerCase().includes(search.toLowerCase())
            )
            .map((student, index) => (
            <tr
              key={index}
              onClick={() => handleStudentClick(student)}
              className={selectedStudent === student ? 'selected-row' : 'event-row'}
            >
              <td>{index + 1}</td>
              <td>{student.firstName}</td>
              <td>{student.indexNumber}</td>
              <td>{departmentList[student.departmentId - 1]}</td>
              <td>{student.semester}</td>
              <td>
                <button onClick={() => handleUpdateStudent(student)} className="EditButton">
                  <FaEdit className="EditIcon" />
                </button>
              </td>
            </tr>
          ))}
        </tbody> : <tbody className='nodata'>
          No Data Available
        </tbody>}
      </table>
    </div>
  );
};

export default StudentTable;
