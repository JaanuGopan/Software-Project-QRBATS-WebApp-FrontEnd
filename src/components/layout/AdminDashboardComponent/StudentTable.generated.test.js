import renderer from 'react-test-renderer';
import React, { useState, useEffect } from "react";

import { FaEdit } from "react-icons/fa";
import GetAllStudentsService from "../../../api/services/GetAllStudentService";
import StudentTable from "./StudentTable";

jest.mock("../../../pages/AdminDashboard/AdminDashboard.css");
jest.mock("react-icons/fa");
jest.mock("../../../api/services/GetAllStudentService");

const renderTree = tree => renderer.create(tree);
describe('<StudentTable>', () => {
  it('should render component', () => {
    expect(renderTree(<StudentTable 
    />).toJSON()).toMatchSnapshot();
  });
  
});