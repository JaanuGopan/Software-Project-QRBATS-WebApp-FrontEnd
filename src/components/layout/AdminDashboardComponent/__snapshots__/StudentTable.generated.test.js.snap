// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StudentTable> should render component 1`] = `
<div
  className="tableDesign"
>
  <table
    className="student-tableArrangement"
  >
    <thead>
      <tr>
        <th>
          No
        </th>
        <th
          className="expand"
        >
          Name
        </th>
        <th>
          IndexNo
        </th>
        <th>
          Department
        </th>
        <th>
          Username
        </th>
        <th>
          Semester
        </th>
        <th>
          Edit
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
